### DISCUSSION TITLE
In for-range aContainer may be not a anonymous copy

### DISCUSSION THREAD

**Discussion by Bububuger** [go101/go101#183] [General]
Hi, I'm studying go101 website, which gives me a lot of inspiration.

In https://go101.org/article/container.html, about `for-range` section
when `aContainer` is an arrary, modification about `aContainer` has effects on it.

```go
package main

import "fmt"

func main() {
	type Person struct {
		name string
		age  int
	}
	persons := [2]Person {{"Alice", 28}, {"Bob", 25}}
	for i, p := range persons {
		fmt.Println(i, p)

		// Now this modification has effects
		// on the iteration.
		persons[1].name = "Jack"

		// This modification has not effects on
		// the persons array, for p is just a
		// copy of a copy of one persons element.
		p.age = 31
	}
	fmt.Println("persons:", &persons)
}
```

output is :
```
0 {Alice 28}
1 {Jack 25}
persons: &[{Alice 28} {Jack 25}]
```

and i tried to print the addresses of the origin `persons` out of `for-range` and the 'copyed' `persons` in `for-range` , they are the same!

by the way, i'm using the go1.14

**Comment by zigo101** [comment]
I get the following output:
```
0 {Alice 28}
1 {Bob 25}
persons: &[{Alice 28} {Jack 25}]
```

What does the `go env` output on your machine?

**Comment by Bububuger** [comment]
ooooh, i am so sorry!   i got it wrong.
output is:
```
0 {Alice 28}
1 {Bob 25}
persons: &[{Alice 28} {Jack 25}]
```
there is no problem this section.
thanks so much for replying.
